test/TestApps/PushNotificationsWin32App/main.cpp

```diff
@@ -1,12 +1,70 @@
 ﻿#include "pch.h"
 #include <iostream>
+#include <sstream>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Security.Cryptography.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <winrt/Windows.Web.Http.Headers.h>
+
 using namespace winrt;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::Storage::Streams;
 using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::ApplicationModel::Activation;
 using namespace winrt::Microsoft::ProjectReunion;
+using namespace Windows::Web::Http;
+
+std::wstring RequestChannel()
+{
+    return // Insert ChannelUri here.;
+}
+
+std::wstring BuildNotificationPayload()
+{
+    std::wstring channel = RequestChannel();
+    std::wstring channelUri = L"\"ChannelUri\":\"" + channel + L"\"";
+    std::wstring x_wns_type = L"\"X_WNS_Type\": \"wns/raw\"";
+    std::wstring contentType = L"\"Content_Type\": \"application/octet-stream\"";
+    std::wstring payload = L"\"Payload\": \"<toast></toast>\"";
+    std::wstring delay = L"\"Delay\": \"false\"";
+    return { L"{" + channelUri + L"," + x_wns_type + L"," + contentType + L"," + payload + L"," + delay + L"}" };
+}
+
+void SendRequestToServer()
+{
+    HttpResponseMessage httpResponseMessage;
+    std::wstring httpResponseBody;
+    try {
+        // Construct the HttpClient and Uri. This endpoint is for test purposes only.
+        HttpClient httpClient;
+        Uri requestUri{ L"https://notificationtestappserver.azurewebsites.net/api/PostPushNotification?" };
+
+        // Construct the JSON to post.
+        HttpStringContent jsonContent(
+            BuildNotificationPayload(),
+            UnicodeEncoding::Utf8,
+            L"application/json");
+
+        // Post the JSON, and wait for a response.
+        httpResponseMessage = httpClient.PostAsync(
+            requestUri,
+            jsonContent).get();
+
+        // Make sure the post succeeded, and write out the response.
+        httpResponseMessage.EnsureSuccessStatusCode();
+        httpResponseBody = httpResponseMessage.Content().ReadAsStringAsync().get();
+        std::wcout << httpResponseBody.c_str() << std::endl;
+    }
+    catch (winrt::hresult_error const& ex)
+    {
+        std::wcout << ex.message().c_str();
+    }
+}
 
 int main()
 {
-    // Add your code here.
-    std::cout << "Hello World!\n";
+    bool succeeded = true;
+    SendRequestToServer();
+    std::getchar();
```

Comment:
I guess this is to have the time to verify the server response?

Reply:
This is a bit strange. Why are we waiting for input here?

Reply:
I think they just want to delay the program exit to see the how the http response output looks like.

We can have a sleep timer there, or say "press a button to exit", but let's not just check in the getchar() trick 😁.

Reply:
When running the application with 'Local Windows Debugger', once the process has completed main(), it displays the error code and waits for user input to close the window. If we were to run the application as packaged and installed on our local machine and run it as a program, it will complete the main(), then kill itself but we aren't able to see any output. std::getchar() allows us to see what succeeded/failed.

Reply:
So couple of things here:
1) The function should wait till the http response comes back and log that either in command line or UI depending on how this process is called.
2) Main should return per-determined error codes so that the caller knows that we failed or succeeded.

Can we achieve 1) and 2)?

Reply:
Again it comes down to how this tool is going to be used for automation scenarios. If it is infact being used for automation, I would get rid of the std::getchar() since it doesnt; really do much (outside of debuggability) and that problem can be solved by the solutions above.